src/EntityFramework6.Npgsql/SqlGenerators/SqlBaseGenerator.cs

```diff
@@ -1160,6 +1167,77 @@ VisitedExpression VisitFunction(EdmFunction function, IList<DbExpression> args,
 #endif
         }
 
+#if ENTITIES6
+        VisitedExpression VisitMatchRegex(EdmFunction function, IList<DbExpression> args, TypeUsage resultType)
+        {
+            if (args.Count != 2 && args.Count != 3)
+                throw new ArgumentException("Invalid number of arguments. Expected 2 or 3.", nameof(args));
+
+            var options = RegexOptions.None;
+
+            if (args.Count == 3)
+            {
+                var optionsExpression = args[2] as DbConstantExpression;
+                if (optionsExpression == null)
+                    throw new NotSupportedException("Options must be constant expression.");
+
+                options = (RegexOptions)optionsExpression.Value;
+            }
+
+            const RegexOptions UnsupportedRegexOptions = RegexOptions.RightToLeft | RegexOptions.ECMAScript;
```

Comment:
Nitpick: I find this const a bit out of place, it is not referred to anywhere else and the exception thrown directly below it clarifies it, just inline the variable.